pkg/controller/rollout/rollout_controller.go

```diff
@@ -124,18 +128,19 @@ func (r *RolloutReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 	// update rollout status
-	done, err := r.updateRolloutStatus(rollout)
+	retry, err := r.updateRolloutStatus(rollout)
```

Comment:
consider move updateRolloutStatus to the end of reconcile function (one can calculate part of new status in the beginning , but update status should be in the end